### ISSUE TITLE
Bookmark Highlights missing metadata

### ISSUE THREAD

**Issue by rlr** [mozilla/activity-stream#1969]
I am on the latest dev version: 1.2.0-dev-1484268155

This is my normal profile that I use daily. But I haven't browsed much in the last ~3-4 days.

<img width="740" alt="screen shot 2017-01-17 at 10 00 57 am" src="https://cloud.githubusercontent.com/assets/36629/22033137/2dde67b2-dc9c-11e6-8bed-887ae97e3be3.png">

**Comment by rlr** [comment]
After restarting browser and updating AS to 1.3.0-dev-1484609321 I am getting 8 highlights. closing for now.

**Comment by jaredkerim** [comment]
Just saw this on another browser, reopening.  Looks like it's falling back to bookmarks before it updates.  I want to understand this behaviour better before we dismiss it.

**Comment by Mardak** [comment]
![screen shot 2017-01-19 at 11 31 29 am](https://cloud.githubusercontent.com/assets/438537/22122101/6fa7bfae-de3b-11e6-8ea3-b15e04de90d2.png)

**Comment by jaredkerim** [comment]
The bookmarks being shown are missing their metadata, we need to make sure that we fetch the metadata for bookmarks the way we do for other URLs.

**Comment by jaredkerim** [comment]
We added logic that boosts the score of bookmarks: https://github.com/mozilla/activity-stream/issues/1887 

However in doing so we also bypassed the logic that prevents us from showing highlights before their metadata has been processed, by setting the weight to non-zero number:

https://github.com/mozilla/activity-stream/blob/master/common/recommender/Baseline.js#L113

In the case where a bookmark does have a suitable preview image, we should wait for the metadata to be processed before we show it so that we don't show a bookmark that would have an image if the metadata was finished processing.  This can be done by changing the + to a * in that line above.  This will still lead to a bookmark being shown 'immediately', but just waiting for the image to be attached before we do.

In the case where a bookmark has no preview image, even after metadata is processed, then we have to figure out whether we want to show it, and if we do, maybe we can use the screenshot as its preview image as is mentioned:

https://github.com/mozilla/activity-stream/issues/2065
